.github/workflows/check-compilers.yml

```diff
@@ -13,15 +13,15 @@ jobs:
     continue-on-error: true
     strategy:
       matrix:
-        cxx: ['g++', 'clang++-13']
+        cxx: ['g++', 'clang++-15']
```

Comment:
Any reason we don't want to use the latest Clang 16?

Reply:
Not specifically. 
his is the clang that's shipped with the OS of the  CI image so the installation (script) was "cleaner".
I suggest that we bump this when we bump the CI OS at some point.